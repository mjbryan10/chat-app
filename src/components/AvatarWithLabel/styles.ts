import styled, { ThemeProps, Theme } from 'styled-components';
import { ThemeColor } from 'shared/theme/types';

interface LabelProps extends ThemeProps<Theme> {
   selected: boolean;
   color?: ThemeColor;
}

export const Label = styled.div<LabelProps>`
   display: flex;
   flex-flow: row wrap;
   justify-content: flex-start;
   align-items: center;
   color: ${({ theme, color }) => {
      if (color) return theme.color[color] || 'inherit';
      return theme.color.text || 'inherit';
   }};
   & h3 {
      margin: 0 auto;
   }
`;
