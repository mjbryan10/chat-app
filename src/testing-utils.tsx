import React from 'react';
import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import { dark } from './shared/theme';

const AllTheProviders = ({ children }: any) => {
  return (
    <ThemeProvider theme={dark}>
        {children}
    </ThemeProvider>
  )
  
}

const customRender = (ui?: any, options?: any) =>
  render(ui, { wrapper: AllTheProviders, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }