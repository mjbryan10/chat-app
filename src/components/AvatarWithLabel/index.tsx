import React, { FC } from 'react';
import * as S from './styles';
import Avatar, { Props as AvatarProps } from '../Avatar';

interface Props extends AvatarProps {
  selected?: boolean;
}

const AvatarWithLabel: FC<Props> = ({
   text,
   size = 50,
   color = 'owner',
   selected = false,
}) => {
  
   return (
      <S.Label selected={selected}>
         <Avatar text={text} size={size} color={color} />
         <span>{text}</span>
      </S.Label>
   );
};

export default AvatarWithLabel;
