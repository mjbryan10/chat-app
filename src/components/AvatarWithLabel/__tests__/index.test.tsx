import React from 'react';
import { render } from 'testing-utils';
import AvatarWithLabel from '..';

describe('AvatarWithLabel component', () => {
   test('should render label if provided in props', () => {
      const testText = 'test label text';
      const {getByText} = render(<AvatarWithLabel title={testText} />);

      expect(getByText(testText)).toBeVisible();
   });
   test('should render label even if no label text provided', () => {
      const {getByTestId} = render(<AvatarWithLabel />);

      expect(getByTestId('avatar-inner-container')).toBeVisible();
   });
   
   test('should render correctly', () => {
      const { asFragment } = render(<AvatarWithLabel />);

      expect(asFragment()).toMatchSnapshot();
   });
});
