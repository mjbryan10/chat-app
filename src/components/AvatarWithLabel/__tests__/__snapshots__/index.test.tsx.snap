// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarWithLabel component should render correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm mBcIY"
    color="owner"
  >
    <div
      class="sc-AxirZ bxDTSS"
      color="owner"
      size="50"
    >
      <span
        class="sc-AxiKw rRURm"
        data-testid="avatar-inner-container"
        size="50"
      >
        A
      </span>
    </div>
    <h3
      data-testid="avatar-with-label-span"
    />
  </div>
</DocumentFragment>
`;
